projects/opendr_ws_2/src/opendr_perception/README.md

```diff
@@ -0,0 +1,874 @@
+A ROS2 node for performing hand gesture recognition using a MobileNetv2 model trained on HANDS dataset.
+The node has been tested with Kinectv2 for depth data acquisition with the following drivers: https://github.com/OpenKinect/libfreenect2 and https://github.com/code-iai/iai_kinect2.
+
+You can find the RGBD hand gesture recognition ROS2 node python script [here](./opendr_perception/rgbd_hand_gesture_recognition_node.py) to inspect the code and modify it as you wish to fit your needs.
+The node makes use of the toolkit's [hand gesture recognition tool](../../../../src/opendr/perception/multimodal_human_centric/rgbd_hand_gesture_learner/rgbd_hand_gesture_learner.py)
+whose documentation can be found [here](../../../../docs/reference/rgbd-hand-gesture-learner.md).
+
+#### Instructions for basic usage:
+
+1. Start the node responsible for publishing images from an RGBD camera. Remember to modify the input topics using the arguments in step 2 if needed.
+
+2. You are then ready to start the hand gesture recognition node:
+    ```shell
+    ros2 run opendr_perception rgbd_hand_gesture_recognition_node.py
+    ```
+    The following optional arguments are available:
+   - `-h or --help`: show a help message and exit
+   - `-ic or --input_rgb_image_topic INPUT_RGB_IMAGE_TOPIC`: topic name for input RGB image (default=`/kinect2/qhd/image_color_rect`)
+   - `-id or --input_depth_image_topic INPUT_DEPTH_IMAGE_TOPIC`: topic name for input depth image (default=`/kinect2/qhd/image_depth_rect`)
+   - `-o or --output_gestures_topic OUTPUT_GESTURES_TOPIC`: topic name for predicted gesture class (default=`/opendr/gestures`)
+   - `--device DEVICE`: device to use, either `cpu` or `cuda`, falls back to `cpu` if GPU or CUDA is not found (default=`cuda`)
+
+3. Default output topics:
+   - Detection messages:`/opendr/gestures`
+
+   For viewing the output, refer to the [notes above.](#notes)
+
+----
+## RGB + Audio input
+
+### Audiovisual Emotion Recognition ROS2 Node
+
+You can find the audiovisual emotion recognition ROS2 node python script [here](./opendr_perception/audiovisual_emotion_recognition_node.py) to inspect the code and modify it as you wish to fit your needs.
+The node makes use of the toolkit's [audiovisual emotion recognition tool](../../../../src/opendr/perception/multimodal_human_centric/audiovisual_emotion_learner/avlearner.py),
+whose documentation can be found [here](../../../../docs/reference/audiovisual-emotion-recognition-learner.md).
+
+#### Instructions for basic usage:
+
+1. Start the node responsible for publishing images. If you have a USB camera, then you can use the `usb_cam_node` as explained in the [prerequisites above](#prerequisites).
+2. Start the node responsible for publishing audio. Remember to modify the input topics using the arguments in step 2 if needed.
+3. You are then ready to start the face detection node
+
+    ```shell
+    ros2 run opendr_perception speech_command_recognition_node.py
```

Comment:
Getting: `ModuleNotFoundError: No module named 'audio_common_msgs'`

Reply:
Thanks for fixing this, the audio package needs to be installed. Maybe we can include it similar to usb cam? I didn't go in this direction because it is needed by only two nodes. The instructions that worked for me can be found [here](https://github.com/opendr-eu/opendr/pull/340#issuecomment-1280625354)

Reply:
ah, wasn't aware of that, I just checked online and it didn't seem that the package existed for ros2

Reply:
Do you think we should include these instructions in the main readme, similar to how usb cam was before we included it in the main installation, or leave it as-is?

Reply:
I think the install procedure (point 1 and 2) should be put in the install.sh script, and point (3) should indeed be mentioned in the documentation as we do for others (like pointcloud etc). As for current commit I think it can be reverted, this is more proper

Reply:
Alright, just to double-check you mean this commit https://github.com/opendr-eu/opendr/pull/256/commits/dae3cde12e1414a4c506fa37611f05b38086c9b9 ?

Reply:
Same stuff should probably be added for ROS1, should we go ahead and do it here?